runtime/near-vm-runner/src/errors.rs

```diff
@@ -65,10 +65,31 @@ impl IntoVMError for wasmer_runtime::error::ResolveError {
 
 impl IntoVMError for wasmer_runtime::error::RuntimeError {
     fn into_vm_error(self) -> VMError {
+        use near_vm_errors::WasmTrap::{BreakpointTrap, GenericTrap};
+        use wasmer_runtime::error::InvokeError;
         use wasmer_runtime::error::RuntimeError;
         match &self {
-            RuntimeError::Trap { msg: _ } => VMError::FunctionCallError(WasmUnknownError),
-            RuntimeError::Error { data } => {
+            RuntimeError::InvokeError(invoke_error) => match invoke_error {
+                InvokeError::FailedWithNoError => VMError::FunctionCallError(WasmUnknownError),
```

Comment:
Would be super helpful if we could comment on each of these arms explaining in what cases they occur so it is clear from reading the code that we exhaustively handle all possible deterministic and non-deterministic cases correctly.

Reply:
Will try to document it per my understanding, but contract/docs on the Wasmer side would be cool.